supervisor/utils/__init__.py

```diff
@@ -36,18 +35,12 @@ async def wrap_api(api, *args, **kwargs):
     return wrap_api
 
 
-def check_port(address: IPv4Address, port: int) -> bool:
-    """Check if port is mapped."""
-    sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
-    sock.settimeout(0.5)
+async def check_port(address: IPv4Address, port: int) -> bool:
+    """Check if port is mapped asynchronously with a timeout."""
     try:
-        result = sock.connect_ex((str(address), port))
-        sock.close()
-
-        # Check if the port is available
-        if result == 0:
-            return True
-    except OSError:
+        await asyncio.wait_for(asyncio.open_connection(str(address), port), 0.5)
```

Comment:
`wait_for` is not recommended because it has some bad cancellation semantics

Reply:
cpython 3.12 will change the underlying implementation of `asyncio.wait_for` to use `asyncio.wait` but that is still a long way off for many people:

https://github.com/python/cpython/pull/98518

Reply:
I see, thanks for the insight! Used your suggestion instead.